x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plan/physical/ExchangeSinkExecSerializationTests.java

```diff
@@ -91,9 +91,45 @@ public void testManyTypeConflictsWithParent() throws IOException {
          */
     }
 
+    private void testManyTypeConflicts(boolean withParent, ByteSizeValue expected) throws IOException {
+        EsIndex index = EsIndexSerializationTests.indexWithManyConflicts(withParent);
+        testSerializePlanWithIndex(index, expected);
+    }
+
+    /**
+     * Test the size of serializing a plan like
+     * FROM index | LIMIT 10
+     * with a single root field that has many children, grandchildren etc.
+     */
+    public void testDeeplyNestedFields() throws IOException {
+        ByteSizeValue expected = ByteSizeValue.ofBytes(47252411);
```

Comment:
On main, this results in a size of 48223371bytes (so, around 1MB more than here).

The `EsRelation` that is serialized as part of the logical plan contains the whole `EsIndex`. Once we stop serializing that `EsIndex` (since it seems unused on the data nodes), I expect the serialized size here to drop further.

C.f. https://github.com/elastic/elasticsearch/issues/112998